### PR TITLE
Support textDocument/documentColor and textDocument/colorPresentation

### PR DESCRIPTION
**Pull request by renkun-ken**
This PR adds support for `textDocument/documentColor` and `textDocument/colorPresentation`.

image

### CODE REVIEW

**Comment by randy3k** [comment]
Wow, I never expect that `colorProvider` will get developed.

As we are enabling more providers, do we want to allow users to disable some of the providers? I don't know how other servers handle it, but I would imagine that we could read options from .Rprofile to disable specific providers.

**Comment by renkun-ken** [comment]
Disabling some providers via `options(...)` in `.Rprofile` sounds perfectly okay.

File: R/handlers-langfeatures.R [hunk 1]
```diff
@@ -175,15 +175,32 @@ document_link_resolve  <- function(self, id, params) {
 #' Handler to the `textDocument/documentColor` [Request].
 #' @keywords internal
 text_document_document_color  <- function(self, id, params) {
-
+    textDocument <- params$textDocument
+    uri <- textDocument$uri
+    document <- self$documents$get(uri)
+    reply <- document_color_reply(id, uri, self$workspace, document)
+    if (is.null(reply)) {
+        queue <- self$pending_replies$get(uri)[["textDocument/documentColor"]]
```
Comment (randy3k) [inline R/handlers-langfeatures.R]: I guess I would work on the pending mechanism at some point. The current implementation is a bit hacky.

Comment (renkun-ken) [reply R/handlers-langfeatures.R]: Sure, glad to see a nicer implementation :)

**Pull request merged**